src/Concordium/Client/Runner.hs

```diff
@@ -1851,6 +1851,31 @@ data ModuleDeployTransactionCfg =
 moduleDeployTransactionPayload :: ModuleDeployTransactionCfg -> Types.Payload
 moduleDeployTransactionPayload ModuleDeployTransactionCfg {..} = Types.DeployModule mdtcModule
 
+-- |Checks if the he given receive name is valid and if so, returns it back
+-- or otherwise a fallback receive name for v1 contracts.
+checkAndGetContractReceiveName :: CI.ContractInfo -> Text -> IO Text
+checkAndGetContractReceiveName contrInfo receiveName = do
+  if CI.hasReceiveMethod receiveName contrInfo
+    then return receiveName
+    else
+      if not $ CI.hasFallbackReceiveSupport contrInfo
+        then confirmAbortOrContinue False
+        else let (_, fallbackReceiveName) = Wasm.contractAndFunctionName $ Wasm.makeFallbackReceiveName $ Wasm.ReceiveName (CI.getContractName contrInfo <> receiveName) in
+          if CI.hasReceiveMethod fallbackReceiveName contrInfo
+          then do
+            logInfo [[i|The contract does not have an entrypoint named '#{receiveName}'. The fallback entrypoint will be used instead.|]]
+            return fallbackReceiveName
+          else confirmAbortOrContinue True
+  where 
+    confirmAbortOrContinue hasFallbackReceiveSupport = do
+      let fallback_msg :: String = if hasFallbackReceiveSupport
+                then " nor a fallback entrypoint"
+                else ""
+      logWarn [[i|The contract has neither an entrypoint named '#{receiveName}'#{fallback_msg}.|]]
```

Comment:
This is not exactly the right message. If `hasFallbackReceiveSupport = False` then the message will be
```
The contract has neither an entrypoint named `foo`.
```
which is not grammatically correct.